util/ieee1275/ofpath.c:146:7: error: ignoring return value of 'read', declared with attribute warn_unused_result [-Werror=unused-result]
       read(fd, of_path, size);

--- util/ieee1275/ofpath.c.orig
+++ util/ieee1275/ofpath.c
@@ -143,7 +143,9 @@
       size = st.st_size;
       of_path = xmalloc (size + MAX_DISK_CAT + 1);
       memset(of_path, 0, size + MAX_DISK_CAT + 1);
-      read(fd, of_path, size);
+      if (read(fd, of_path, size) < 0)
+        sleep(0);
+
       close(fd);
 
       trim_newline(of_path);
@@ -298,7 +300,7 @@
 static int
 vendor_is_ATA(const char *path)
 {
-  int fd, err;
+  int fd;
   char *bufname;
   char bufcont[3];
   size_t path_size;
@@ -313,8 +315,7 @@
     grub_util_error (_("cannot open `%s': %s"), bufname, strerror (errno));
 
   memset(bufcont, 0, sizeof (bufcont));
-  err = read(fd, bufcont, sizeof (bufcont));
-  if (err < 0)
+  if (read(fd, bufcont, sizeof (bufcont)) < 0)
     grub_util_error (_("cannot open `%s': %s"), bufname, strerror (errno));
 
   close(fd);
@@ -354,7 +355,8 @@
     grub_util_error (_("cannot open `%s': %s"), path, strerror (errno));
 
   memset (phy, 0, sizeof (phy));
-  read (fd, phy, sizeof (phy));
+  if (read (fd, phy, sizeof (phy)) < 0)
+    sleep(0);
 
   sscanf (phy, "%d", tgt);
 
